sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -79,8 +79,15 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
   }
 
   /**
-   * Register a user-defined function (UDF), for a UDF that's already defined using the DataFrame
-   * API (i.e. of type UserDefinedFunction).
+   * Registers a user-defined function (UDF), for a UDF that's already defined using the Dataset
+   * API (i.e. of type UserDefinedFunction). To change a UDF to nondeterministic, call the API
+   * `UserDefinedFunction.asNondeterministic()`.
```

Comment:
a good example will be
```
val foo = udf(() => { "hello" })
spark.udf.register("stringContant", foo.asNonNullable())
```
Although the return type of the UDF is `String` and is nullable, but we know that this UDF will never return null.